ttnn/cpp/ttnn/operations/normalization/layernorm/device/layernorm_op.hpp

```diff
@@ -0,0 +1,95 @@
+#include "tt_eager/tensor/tensor.hpp"
+#include "ttnn/operations/core.hpp"
+
+using namespace tt::constants;
+
+namespace ttnn::operations::normalization {
+
+enum class LayerNormType {
+    LAYERNORM, RMSNORM
+};
+
+struct LayerNormDefaultProgramConfig{
+    static constexpr auto attribute_names = std::forward_as_tuple();
+    static constexpr auto attribute_values() { return std::forward_as_tuple(); }
+};
+struct LayerNormShardedMultiCoreProgramConfig {
+    CoreCoord compute_with_storage_grid_size;
+    std::size_t subblock_w;
+    std::size_t block_h;
+    std::size_t block_w;
+    bool inplace;
+
+    static constexpr auto attribute_names =
+        std::forward_as_tuple("compute_with_storage_grid_size", "subblock_w", "block_h", "block_w", "inplace");
+
+    const auto attribute_values() const {
+        return std::forward_as_tuple(compute_with_storage_grid_size, subblock_w, block_h, block_w, inplace);
+    }
+};
+
+using LayerNormProgramConfig = std::variant<
+    LayerNormDefaultProgramConfig,
+    LayerNormShardedMultiCoreProgramConfig
+>;
+
+operation::ProgramWithCallbacks layernorm_multi_core(
+    const Tensor &a,
+    const std::optional<const Tensor> b,
+    const std::optional<const Tensor> gamma,
+    const std::optional<const Tensor> beta,
+    Tensor& output,
+    LayerNormType norm_type,
+    float eps,
+    DeviceComputeKernelConfig compute_kernel_config
+);
+
+operation::ProgramWithCallbacks layernorm_multi_core_sharded(
+    const Tensor &a,
+    const std::optional<const Tensor> b,
+    const std::optional<const Tensor> gamma,
+    const std::optional<const Tensor> beta,
+    Tensor& output,
+    LayerNormType norm_type,
+    float eps,
+    CoreCoord compute_grid_size,
+    uint32_t subblock_wt,
+    uint32_t block_ht,
+    uint32_t block_wt,
+    DeviceComputeKernelConfig compute_kernel_config
+);
+
+struct LayerNorm {
+    LayerNormType norm_type;
+    float eps;
+    MemoryConfig output_mem_config;
+    LayerNormProgramConfig program_config;
+    const DeviceComputeKernelConfig compute_kernel_config;
+
+    void validate(const std::vector<Tensor> &input_tensors, const std::vector<std::optional<const Tensor>>& optional_input_tensors) const;
+    std::vector<tt::tt_metal::Shape> compute_output_shapes(const std::vector<Tensor> &input_tensors) const;
+    std::vector<Tensor> create_output_tensors(const std::vector<Tensor> &input_tensors) const;
+    operation::ProgramWithCallbacks create_program(
+        const std::vector<Tensor>& input_tensors,
+        const std::vector<std::optional<const Tensor>>& optional_input_tensors,
+        std::vector<Tensor> &output_tensors
+    ) const;
+
+    static constexpr auto attribute_names =
+        std::forward_as_tuple("norm_type", "eps", "output_mem_config", "program_config", "compute_kernel_config");
+    const auto attribute_values() const {
+        return std::forward_as_tuple(norm_type, eps, output_mem_config, program_config, compute_kernel_config);
+    }
```

Comment:
I think this is not needed after last change from @arakhmati 
https://github.com/tenstorrent/tt-metal/commit/a0dbfe07f6726421ea90eeb02fe3548b9e5fee83

Reply:
will remove